src/application/mod.rs

```diff
@@ -102,6 +106,14 @@ impl Default for Application {
 impl Application {
     /// Creates a new Application using the given ApplicationConf
     fn new(conf: ApplicationConf) -> Self {
+        #[cfg(feature = "metrics")]
```

Comment:
As mentioned in the comment. Default to a LogRecorder if only the metrics flag is enabled. And have prometheus feature under ``prometheus_exporter``.